Automattic/wp-calypso
# Media: Manage modal current view using Redux state

Related: #8625

This pull request seeks to migrate the post editor media modal to use Redux state to manage the currently active view of the modal. This will be used as part of the Site Icon project to allow us to control the behavior of the confirmation action, in order to send users through the flow of cropping an image after making a selection. This is also a general refactoring of the modal, relying on global Redux state rather than the modal's component state to maintain the selected view.

__Testing Instructions:__

Verify that there is no change in behavior of using the media modal. In particular, test:

- View media detail
- Insert multiple images as gallery
- View media detail when multiple items selected, and verify behavior of deleting one item
- Use the image editor
- Selecting a gallery and clicking the Edit inline toolbar should launch the modal to the gallery view
- "Back to Media Library"
- Cancelling modal should reset to list next time modal is opened

/cc @gwwar @artpi @lamosty 

P.S.: Sorry the size of this grew a little out of control. It seemed like it was going to be a small task 😄
